packages/compiler-cli/test/ngtsc/authoring_spec.ts

```diff
@@ -0,0 +1,192 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {runInEachFileSystem} from '../../src/ngtsc/file_system/testing';
+import {loadStandardTestFiles} from '../../src/ngtsc/testing';
+
+import {NgtscTestEnvironment} from './env';
+
+const testFiles = loadStandardTestFiles();
+
+runInEachFileSystem(() => {
+  describe('input()', () => {
+    let env!: NgtscTestEnvironment;
+
+    beforeEach(() => {
+      env = NgtscTestEnvironment.setup(testFiles);
+      env.tsconfig({strictTemplates: true});
+    });
+
+    it('should handle a basic, primitive valued input', () => {
+      env.write('test.ts', `
+        import {Directive, input} from '@angular/core';
+
+        @Directive()
+        export class TestDir {
+          data = input('test');
+        }
+      `);
+      env.driveMain();
+      const js = env.getContents('test.js');
+      expect(js).toContain('inputs: { data: [1, "data"] }');
+    });
+
+    it('should handle an alias configured, primitive valued input', () => {
+      env.write('test.ts', `
+        import {Directive, input} from '@angular/core';
+
+        @Directive()
+        export class TestDir {
+          data = input('test', {
+            alias: 'publicName',
+          });
+        }
+      `);
+      env.driveMain();
+      const js = env.getContents('test.js');
+      expect(js).toContain('inputs: { data: [1, "publicName", "data"] }');
+    });
+
+    it('should error if a required input declares an initial value', () => {
+      env.write('test.ts', `
+        import {Directive, input} from '@angular/core';
+
+        @Directive()
+        export class TestDir {
+          data = input.required({
+            initialValue: 'bla',
+          });
+        }
+      `);
+      const diagnostics = env.driveDiagnostics();
+      expect(diagnostics[0].messageText).toEqual(jasmine.objectContaining({
+        messageText: 'No overload matches this call.',
+      }));
+    });
+
+
+    it('should handle a transform and required input', () => {
+      env.write('test.ts', `
+        import {Directive, input} from '@angular/core';
+
+        @Directive()
+        export class TestDir {
+          data = input.required({
+            transform: (v: string|number) => 'works',
+          });
+        }
+      `);
+      env.driveMain();
+      expect(env.getContents('test.js')).toContain(`inputs: { data: [1, "data"] }`);
+      expect(env.getContents('test.d.ts')).toContain('"required": true; "isSignal": true;');
+      expect(env.getContents('test.d.ts'))
+          .withContext(
+              'Expected no coercion member as input signal captures the write type of the transform')
+          .not.toContain('ngAcceptInputType');
+    });
+
+
+    it('should handle a non-primitive initial values', () => {
```

Comment:
```suggestion
    it('should handle non-primitive initial values', () => {
```

(or drop the plural `s` for `values`)